src/imp_cs.rs

```diff
@@ -0,0 +1,78 @@
+use core::panic::{RefUnwindSafe, UnwindSafe};
+
+use atomic_polyfill::{AtomicBool, Ordering};
+use critical_section::{CriticalSection, Mutex};
+
+use crate::unsync::OnceCell as UnsyncOnceCell;
+
+pub(crate) struct OnceCell<T> {
+    initialized: AtomicBool,
+    value: Mutex<UnsyncOnceCell<T>>,
```

Comment:
On mobile, so not a proper review, but I am wondering why this needs to be a OnceCell? It seems like an Option should be enough perhaps?

Reply:
Ah, the docs explain that nicely:

>It does not provide interior mutability.

worth adding a comment about this!

Reply:
Ah, yes, that too. And also, using `unsync::OnceCell` inside allows re-using the `get_or_try_init` logic.